### PR TITLE
1731 trade input

### PR DESCRIPTION
**Pull request by stephensprinkle**

### CODE REVIEW

**Comment by tinybike** [comment]
Wow, it's 100% less annoying now :grinning:

**Comment by tinybike** [comment]
Getting this warning after sending a trade:
```
warning.js?8a56:36 Warning: `value` prop on `input` should not be null. Consider using the empty string to clear the component or `undefined` for uncontrolled components.
    in input (created by Input)
    in div (created by Input)
    in Input (created by OutcomeTrade)
    in div (created by OutcomeTrade)
    in div (created by OutcomeTrade)
    in article (created by OutcomeTrade)
    in OutcomeTrade (created by Outcome)
    in article (created by Outcome)
    in Outcome (created by Outcomes)
    in div (created by Outcomes)
    in article (created by Outcomes)
    in Outcomes (created by MarketData)
    in article (created by MarketData)
    in MarketData (created by MarketActive)
    in div (created by MarketActive)
    in article (created by MarketActive)
    in MarketActive (created by MarketView)
    in section (created by MarketView)
    in MarketView
    in div (created by Routes)
    in Routes (created by App)
    in div (created by App)
    in div (created by App)
    in div (created by App)
    in div (created by App)
    in main (created by App)
    in App
    in AppContainer
```
Possibly due to setting state to `null` instead of an empty string?

File: src/modules/outcomes/components/outcome-trade.jsx [hunk 1]
```diff
@@ -93,18 +89,14 @@ export default class OutcomeTrade extends Component {
 	}
 
 	validatePrice(value, trade) {
-		let isLimitPriceValueValid;
-
 		if (value != null) {
 			if ((value >= parseFloat(this.state.minLimitPrice) && value <= parseFloat(this.state.maxLimitPrice)) || value === '') {
-				isLimitPriceValueValid = true;
+				this.setState({ isLimitPriceValueValid: true });
 				trade.updateTradeOrder(null, value, trade.side);
```
Comment (tinybike) [inline src/modules/outcomes/components/outcome-trade.jsx]: Should this be `trade.updateTradeOrder('', value, trade.side);` instead?

**Comment by tinybike** [comment]
Forgot to add -- this error also occurs alongside the one pasted above:
```
warning.js?8a56:36 Warning: Input is changing a controlled input of type undefined to be uncontrolled. Input elements should not switch from controlled to uncontrolled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://fb.me/react-controlled-components
```

**Comment by stephensprinkle** [comment]
Just updated the PR w/ a fix to the trade component.  

Ended up being an issue w/ the trade component’s inputs not re-rendering once the trade orders were cleared.  This was occurring previously, but I’d refactored out the check for the share + price inputs so that focus wouldn’t be lost...side-effect being the components weren’t re-rendered upon `tradeOrders` clearing on `submitTrade`.  

Sorry about the delay, with the faucet acting up I had to wait to re-sync my node to testnet.

**Pull request merged**